qiskit/circuit/library/arithmetic/linear_pauli_rotations.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Linearly-controlled X, Y or Z rotation."""
+
+from typing import Optional
+
+from qiskit.circuit import QuantumRegister
+from qiskit.circuit.exceptions import CircuitError
+
+from .functional_pauli_rotations import FunctionalPauliRotations
+
+
+class LinearPauliRotations(FunctionalPauliRotations):
+    r"""Linearly-controlled X, Y or Z rotation.
+
+    For a register of state qubits |x>, a target qubit |0> and the basis 'Y' this
+    circuit acts as:
+
+    q_0:     |0>─────────────────────────■───────── ... ──────────────────────
+                                         │
+                                         .
+                                         │
+    q_(n-1): |0>─────────────────────────┼───────── ... ───────────■──────────
+                 ┌────────────┐  ┌───────┴───────┐       ┌─────────┴─────────┐
+    q_n:     |0>─┤ RY(offset) ├──┤ RY(2^0 slope) ├  ...  ┤ RY(2^(n-1) slope) ├
+                 └────────────┘  └───────────────┘       └───────────────────┘
```

Comment:
this general circuit form diagram is super useful, can you add it for all of them? Also I think it may render better if you put it in a `.. parsed-literal::` block (that's what i did with the gate documentation), but i'm not sure. Also you have to remove `|0>` from the beginning of the circuits as they cause formatting errors (which is good because they shouldn't be there anyway, the drawers are being updated to not assume circuit initial state).